### PR TITLE
Add block device memory leak test

### PR DESCRIPTION
**Pull request by qingwangrh**
ID:2047671
Signed-off-by: qingwangrh <qinwang@redhat.com>

### CODE REVIEW

**Comment by qingwangrh** [comment]
python ConfigTest.py --testcase=block_check_memory_leak --guestname=RHEL.9.0.0 --clone=no --machine=q35,i440fx
(1/2) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.block_check_memory_leak: PASS (1171.52 s)
 (2/2) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.block_check_memory_leak.q35: PASS (1212.23 s)

**Comment by qingwangrh** [comment]
@YongxueHong @zhenyzha Please help to review. thanks.

File: qemu/tests/block_check_memory_leak.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        guest_dir = params["guest_dir"]
+        deps_dir = virttest_data_dir.get_deps_dir()
+        host_file = os.path.join(deps_dir, host_script)
+        vm.copy_files_to(host_file, guest_dir)
+        logger.info("Execute io:%s", guest_io_cmd)
+        session.sendline("$SHELL " + guest_io_cmd)
+
+    logger = logging.getLogger('avocado.run')
+    data_images = params['data_images'].split()
+    error_context.context("Get the main VM", logger.info)
```
[the remark this answers is not in the document]
Comment (qingwangrh) [reply qemu/tests/block_check_memory_leak.py:48]: set logger to test.log

**Comment by zhenyzha** [comment]
git am 3205.patch 
Applying: Add block device memory leak test
.git/rebase-apply/patch:154: trailing whitespace.
    
.git/rebase-apply/patch:163: trailing whitespace.
    
.git/rebase-apply/patch:173: trailing whitespace.
    
.git/rebase-apply/patch:182: trailing whitespace.
    
.git/rebase-apply/patch:187: trailing whitespace.
    
warning: squelched 2 whitespace errors
warning: 7 lines add whitespace errors.

**Comment by zhenyzha** [comment]
The detailed log is uploaded in the bug

(1/1) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.block_check_memory_leak.arm64-pci: ERROR: Could not connect to monitor 'qmpmonitor1' (104.07 s)

**Comment by qingwangrh** [comment]
> The detailed log is uploaded in the bug
> 
> (1/1) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.block_check_memory_leak.arm64-pci: ERROR: Could not connect to monitor 'qmpmonitor1' (104.07 s)

I guess the vm_create_timeout is too small on arm platform, @zhenyzha ,could you help to retry it ?

**Comment by zhenyzha** [comment]
> 

Sure, testing, update results later.

**Comment by zhenyzha** [comment]
Still having a similar issue:

```
02:57:46 WARNI| Could not connect to monitor 'Could not connect to monitor socket: [Errno 2] No such file or directory'
02:57:47 INFO | Connecting to monitor '<<class 'virttest.qemu_monitor.QMPMonitor'>> qmpmonitor1'
02:58:08 WARNI| No QMP greeting message received. Output so far: 
```

**Comment by zhenyzha** [comment]
Since some of the arm machines are old, adding `login_timeout` to 1600 cases can pass.

`(1/1) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.block_check_memory_leak.arm64-pci: PASS (2747.31 s)`

**Comment by qingwangrh** [comment]
> Since some of the arm machines are old, adding `login_timeout` to 1600 cases can pass.
> 
> `(1/1) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.block_check_memory_leak.arm64-pci: PASS (2747.31 s)`

Thanks,updated.

**Review by YongxueHong — APPROVED** [review]

**Pull request merged**